rollup/sync_service/sync_service.go

```diff
@@ -145,13 +175,95 @@ func (s *SyncService) SubscribeNewL1MsgsEvent(ch chan<- core.NewL1MsgsEvent) eve
 	return s.scope.Track(s.msgCountFeed.Subscribe(ch))
 }
 
+func (s *SyncService) SubscribeNewL1BlocksTx(ch chan<- core.NewL1MsgsEvent) event.Subscription {
+	return s.scope.Track(s.l1BlocksPool.l1BlocksFeed.Subscribe(ch))
+}
+
+func (s *SyncService) CollectL1BlocksTxs(latestL1BlockNumberOnL2, maxNumTxs uint64) []*types.SystemTx {
```

Comment:
Will this method be able to handle L2 reorgs? E.g. when we've marked some L1 blocks as included (removed from `pendingL1BlocksTxs`) but then some blocks are reorged.